extra/CMakeLists.txt

```diff
@@ -1,63 +1,79 @@
-set(nmodl_makefile "nrnivmodl_core_makefile")
-set(nmodl_compile_scipt "nrnivmodl-core")
+# =============================================================================
+# Copyright (C) 2016-2019 Blue Brain Project
+#
+# See top-level LICENSE file for details.
+# =============================================================================
 
-get_directory_property(CORENRN_COMPILE_FLAGS COMPILE_DEFINITIONS)
+# =============================================================================
+# Prepare nrnivmodl-core script with correct compiler flags
+# =============================================================================
 
-if (CORENRN_COMPILE_FLAGS)
-    set(CORENRN_COMPILE_DEFS "-D${CORENRN_COMPILE_FLAGS}")
-    string(REPLACE ";" " -D" CORENRN_COMPILE_DEFS "${CORENRN_COMPILE_DEFS}")
+# extract the COMPILE_DEFINITIONS property from the directory.
+get_directory_property(CORENRN_COMPILE_FLAGS COMPILE_DEFINITIONS)
```

Comment:
why do we need the directory scope here?

Reply:
Fernando did this but I assume this is the way to get all compiler flags : https://stackoverflow.com/questions/5403705/cmake-add-definitions-and-compile-definitions-how-to-see-them

Reply:
OK, I did not know that